docs/spec/src/integration/rollup-stacks/2-op-hokulea-secure-integration.md

```diff
@@ -0,0 +1,33 @@
+### Hokulea
+
+Hokulea provides a Rust implementation of EigenDA blob derivation for the OP stack. The Hokulea client (and its associated crates) implements the EigenDA blob derivation logic described in the [EigenDA blob derivation section](#eigenda-blob-derivation). The client is designed to be imported as a library into the OP consensus Rust implementation [Kona](https://github.com/op-rs/kona).
+
+Since the OP rollup inbox is not a smart contract, the secure integration requires EigenDA blob derivation to take place entirely offchain (see the design rationale in [secure integration](../spec/6-secure-integration.md#secure-integration-framework)). Depending on the choice of VM and game type, Hokulea can support optimistic fault proofs, ZK fault proofs, and validity ZK proofs.
+
+
+![](../../assets/integration/hokulea-preimage-derivation-impl.png)
+
+#### Preimage Oracle Architecture
+
+The major difference between implementations lies in how the preimage oracle is abstracted and implemented. In Hokulea, the interface is abstracted as a key-value map to make the preimage oracle verifiable on L1 Ethereum.
+
+The Hokulea preimage host for the key-value oracle interface is built on top of the EigenDA proxy (see diagram above). The proxy handles all the heavy lifting to retrieve the actual preimage data, while the Hokulea host serves as a thin layer that translates HTTP status codes into preimage data or errors.
+
+#### Communication Between Hokulea Host and EigenDA Proxy
+
+The proxy uses an HTTP interface to serve as a base layer for abstraction. The proxy exposes the following JSON status codes in addition to HTTP status codes to convey information about the preimage:
+
+| Message             | HTTP Status Code | JSON Status Code | Indication           |
```

Comment:
```suggestion
The proxy uses an HTTP interface to serve as a base layer for abstraction. The proxy exposes the following app-layer status codes in addition to HTTP status codes to convey information about the preimage:

| Message             | HTTP Status Code | App-Layer Status Code | Indication           |
```

Reply:
not strictly necessary but when I first saw "json status codes" I was confused thinking these were json-rpc status codes.